server/app/services/applicant/ApplicantService.java

```diff
@@ -1153,15 +1158,17 @@ private ApplicationPrograms relevantProgramsForApplicantInternal(
             }
             programNamesWithApplications.add(programName);
           } else if (maybeSubmittedApp.isPresent() && activeProgramNames.containsKey(programName)) {
-            // When extracting the application status, the definitions associated with the program
-            // version at the time of submission are used. However, when clicking "reapply", we use
-            // the latest program version below.
+            // When extracting the application status, the Statuses present in the current program
+            // versions are used
             ProgramDefinition applicationProgramVersion =
                 programRepository.getShallowProgramDefinition(maybeSubmittedApp.get().getProgram());
+            StatusDefinitions activeStatusDefinitions =
+                applicationStatusesRepository.lookupActiveStatusDefinitions(
```

Comment:
Good catch! I will try to move it outside the loop.

Reply:
I checked, the foreach is happening for each program, so for each program present, we are checking if the program is in 1. Not-started state, 2. In incomplete state, 3. Completed state. Since each program is different, we cannot move this call outside the loop.